### PR TITLE
CI/Tests: Optimize GitHub workflows tests performance 

### PR DESCRIPTION
**Pull request by keon94**
### Summary
What does this PR do?
1. Eliminate Go plugin compilation from being needed for the tests. This will massively speed up the builds (from ~30m to < 5m) 
2. Stabilize The Integration test framework to be resilient toward existing DB leftovers from previous test stages
3. Skip failing Python tests until fixed here: https://github.com/apache/incubator-devlake/pull/4840 (Update: no longer skipping since it's merged)

### Does this close any open issues?
Closes n/a

### CODE REVIEW

File: backend/core/migration/migrator.go [hunk 1]
```diff
@@ -84,10 +89,9 @@ func (m *migratorImpl) Execute() errors.Error {
 	})
 	// execute them one by one
 	db := m.basicRes.GetDal()
-	logger := m.basicRes.GetLogger().Nested("migrator")
 	for _, swc := range m.pending {
-		scriptId := fmt.Sprintf("%d-%s", swc.script.Version(), swc.script.Name())
```
Comment (keon94) [inline backend/core/migration/migrator.go:89]: this cache key (%d-%s) was inconsistent with the key format used in the rest of the code (%s:%d). This is a bug, right? @klesh

File: backend/test/e2e/api/task/task_test.go [hunk 2]
```diff
@@ -1,59 +0,0 @@
-/*
```
Comment (keon94) [inline backend/test/e2e/api/task/task_test.go:1]: This was a useless test, hence removed.

File: backend/test/integration/.todo [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/*
```
Comment (keon94) [inline backend/test/integration/.todo:1]: placeholder for future

**Review by klesh — APPROVED** [review]
LGTM

**Pull request merged**